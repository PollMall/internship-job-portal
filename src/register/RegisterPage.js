import React from 'react';
import { Link } from 'react-router-dom';
import { Grid } from '@material-ui/core';
import RegisterForm from './RegisterForm';

function RegisterPage() {
  return (
    <Grid container alignItems="center" direction="column">
      <h1>Register Form</h1>
      <RegisterForm />
      <p>
        Already have an account?
        <Link to="/login">
          Login Now
        </Link>
      </p>
    </Grid>
  );
}

export default RegisterPage;
