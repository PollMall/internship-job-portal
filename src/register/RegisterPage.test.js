import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { screen, render } from '@testing-library/react';
import RegisterPage from './RegisterPage';

describe('RegisterPage test', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: () => ({
        matches: false,
        addListener: () => {},
        removeListener: () => {},
      }),
    });
  });

  it('should match snapshot', () => {
    render(<RegisterPage />, { wrapper: MemoryRouter });
    expect(screen).toMatchSnapshot();
  });
});
