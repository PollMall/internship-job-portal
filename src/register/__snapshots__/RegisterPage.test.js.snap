// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage test should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
  >
    <h1>
      Register Form
    </h1>
    <p>
      Please provide your personal information
    </p>
    <form
      class="makeStyles-root-1"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="username"
          id="username-label"
        >
          Username
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="username"
            name="username"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="firstname"
          id="firstname-label"
        >
          First Name
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="firstname"
            name="firstName"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="lastname"
          id="lastname-label"
        >
          Last Name
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="lastname"
            name="lastName"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
      </div>
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Register
        </span>
      </button>
    </form>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 0; visibility: hidden;"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <p>
      Already have an account?
      <a
        href="/login"
      >
        Login Now
      </a>
    </p>
  </div>
</div>
`;
